packages/expo-image-picker/src/ImagePicker.types.ts

```diff
@@ -295,18 +298,105 @@ export type ImagePickerErrorResult = {
 };
 
 // @needsAudit
+export type ImagePickerResult = {
+  /**
+   * An array of picked assets or `null` when the request was canceled.
+   */
+  assets: ImagePickerAsset[] | null;
+  /**
+   * Boolean flag which shows if request was canceled. If asset data have been returned this should
+   * always be `false`.
+   */
+  canceled: boolean;
+  /**
+   * @hidden
```

Comment:
I also don't feel like we should hide the deprecated field, if we do so, this will be an exception for the pattern.

Hiding those fields removes the vital information for the user how to migrate or which other field or type to use from documentation.